src/Cake.Core/IO/DirectoryPathCollection.cs

```diff
@@ -27,8 +27,8 @@ public sealed class DirectoryPathCollection : IEnumerable<DirectoryPath>
         /// <summary>
         /// Initializes a new instance of the <see cref="DirectoryPathCollection"/> class.
         /// </summary>
-        /// <param name="comparer">The comparer.</param>
-        public DirectoryPathCollection(PathComparer comparer)
+        /// <param name="comparer">The comparer. Defaults to <see cref="P:PathComparer.Default" />.</param>
+        public DirectoryPathCollection(PathComparer comparer = null)
```

Comment:
@cake-build/cake-team please review potential breaking change.

Reply:
When I added `GlobbingAliases.GetPaths` I saw that there was some duplication in the existing code that effectively initialized the `comparer` parameter to `PathComparer.Default` in:
- GlobbingAliases.GetPaths
- GlobbingAliases.GetDirectories
- GlobbingAliases.GetFiles
- DirectoryAliases.GetSubDirectories
- VSWhereTool.RunVSWhere

I thought it would be better to move this default initialization from the call site to the various *`PathCollection` based constructors. I did this using a default parameter, which should be backwards compatible, unless I'm missing something, but it could be also be done with another default ctor overload that calls the existing ctor (and dropping the default parameters). Technically that would be nicer as it simplifies all the call sites. Is there a preferred approach in this repository?

Happy to change it. 😃

Reply:
This will retain API compatibility, but will break  ABI which means any addin or module utilizing this will throw an runtime error. Only non breaking way to add it would be to add a new parameterless contructor that forwards null to the existing construction.

Reply:
@devlead so would you accept the change if I refactor to add a default ctor and drop the default parameter?

Reply:
Probably, would need to review and test more to be certain.